### DISCUSSION TITLE
is GraphQLConfiguration threadsafe?

### DISCUSSION THREAD

**Discussion by eduarddrenth** [graphql-java-kickstart/graphql-java-servlet#380] [Q&A] [answered]
Dear all,

For frysker.nl I use graphql-java-kickstart. I would like to know if i can statically cache GraphQLConfiguration and use it from different threads.

Regards, Eduard

**Comment by oliemansm** [comment, marked as answer]
Hi Eduard,

Not sure when this would apply tbh. Are you creating multiple servlets or something? Shouldn't be a problem though.

**Comment by eduarddrenth** [comment]
As far as I know servlet containers operate using a pool of servlets, so multiple instances. Quote:

The web container manages the lifecycle of servlets (it is responsible for creating and managing instances of servlets). The Java EE specification does not guarantee that there will be only one instance of your servlet class, so you should not write your servlets in a way that depend on this.

**Reply by oliemansm** [reply]
Yeah ok, so just normal servlet usage. That's not a problem then.

**Comment by eduarddrenth** [comment]
So just to be sure: I can statically cache the GraphQLConfiguration?

**Reply by oliemansm** [reply]
yes

**Comment by eduarddrenth** [comment]
Thanks!
